Sources/Datadog/Datadog.swift

```diff
@@ -88,12 +88,16 @@ public class Datadog {
         guard Datadog.instance == nil else {
             throw ProgrammerError(description: "SDK is already initialized.")
         }
-        guard let logsUploadURL = configuration.logsUploadURL else {
-            throw ProgrammerError(description: "SDK configuration is invalid - check `logsEndpoint` and (or) `clientToken`.")
+        guard let logsUploadURLProvider = try? UploadURLProvider(
+            endpointURL: configuration.logsEndpoint.url,
+            clientToken: configuration.clientToken,
+            dateProvider: SystemDateProvider()
+            ) else {
+                throw ProgrammerError(description: "SDK configuration is invalid - check `logsEndpoint` and (or) `clientToken`.")
```

Comment:
Let's just re-throwing the existing errors. Now we capture them with `try?` and erase the error information by throwing another `ProgrammerError`. This would be easier and nicer and shorter:
```swift
let logsUploadURLProvider = try UploadURLProvider(
   endpointURL: configuration.logsEndpoint.url,
   clientToken: configuration.clientToken,
   dateProvider: SystemDateProvider()
)
```

The two distinct errors are already thrown from `UploadURLProvider.init`:

```swift
// UploadURLProvider

init(endpointURL: String, clientToken: String, dateProvider: DateProvider) throws {
   guard !endpointURL.isEmpty, let endpointURL = URL(string: endpointURL) else {
      throw ProgrammerError(description: "`endpointURL` cannot be empty.")
   }
   guard !clientToken.isEmpty else {
      throw ProgrammerError(description: "`clientToken` cannot be empty.")
   }

   // ...
}
```

Reply:
good catch 💯 i missed that while copy&pasting